lib/puppet_x/puppetlabs/scheduled_task/taskscheduler2.rb

```diff
@@ -0,0 +1,318 @@
+  TASK_ACTION_COM_HANDLER = 5
+  TASK_ACTION_SEND_EMAIL = 6
+  TASK_ACTION_SHOW_MESSAGE = 7
+
+  # https://msdn.microsoft.com/en-us/library/windows/desktop/aa383557(v=vs.85).aspx
+  # Undocumented values
+  # Win7/2008 R2                       = 3
+  # Win8/Server 2012 R2 or Server 2016 = 4
+  # Windows 10                         = 6
+  TASK_COMPATIBILITY_AT = 0
+  TASK_COMPATIBILITY_V1 = 1
+  TASK_COMPATIBILITY_V2 = 2
+
+  # https://msdn.microsoft.com/en-us/library/windows/desktop/aa383915%28v=vs.85%29.aspx
+  TASK_TRIGGER_EVENT                 = 0
+  TASK_TRIGGER_TIME                  = 1
+  TASK_TRIGGER_DAILY                 = 2
+  TASK_TRIGGER_WEEKLY                = 3
+  TASK_TRIGGER_MONTHLY               = 4
+  TASK_TRIGGER_MONTHLYDOW            = 5
+  TASK_TRIGGER_IDLE                  = 6
+  TASK_TRIGGER_REGISTRATION          = 7
+  TASK_TRIGGER_BOOT                  = 8
+  TASK_TRIGGER_LOGON                 = 9
+  TASK_TRIGGER_SESSION_STATE_CHANGE  = 11
+
+  # https://msdn.microsoft.com/en-us/library/windows/desktop/aa382538%28v=vs.85%29.aspx
+  TASK_VALIDATE_ONLY                 = 0x1
+  TASK_CREATE                        = 0x2
+  TASK_UPDATE                        = 0x4
+  TASK_CREATE_OR_UPDATE              = 0x6
+  TASK_DISABLE                       = 0x8
+  TASK_DONT_ADD_PRINCIPAL_ACE        = 0x10
+  TASK_IGNORE_REGISTRATION_TRIGGERS  = 0x20
+
+  # https://msdn.microsoft.com/en-us/library/windows/desktop/aa383566(v=vs.85).aspx
+  TASK_LOGON_NONE                           = 0
+  TASK_LOGON_PASSWORD                       = 1
+  TASK_LOGON_S4U                            = 2
+  TASK_LOGON_INTERACTIVE_TOKEN              = 3
+  TASK_LOGON_GROUP                          = 4
+  TASK_LOGON_SERVICE_ACCOUNT                = 5
+  TASK_LOGON_INTERACTIVE_TOKEN_OR_PASSWORD  = 6
+
+  # https://msdn.microsoft.com/en-us/library/windows/desktop/aa380747(v=vs.85).aspx
+  TASK_RUNLEVEL_LUA     = 0
+  TASK_RUNLEVEL_HIGHEST = 1
+
+  def self.folder_path_from_task_path(task_path)
+    path = task_path.rpartition('\\')[0]
+
+    path.empty? ? ROOT_FOLDER : path
+  end
+
+  def self.task_name_from_task_path(task_path)
+    task_path.rpartition('\\')[2]
+  end
+
+  # Returns an array of scheduled task names.
+  # By default EVERYTHING is enumerated
+  # option hash
+  #    include_child_folders: recurses into child folders for tasks. Default true
+  #    include_compatibility: Only include tasks which have any of the specified compatibility levels. Default empty array (everything is permitted)
+  #
+  def self.enum_task_names(folder_path = ROOT_FOLDER, options = {})
+    raise TypeError unless folder_path.is_a?(String)
+
+    options[:include_child_folders] = true if options[:include_child_folders].nil?
+    options[:include_compatibility] = [] if options[:include_compatibility].nil?
+
+    array = []
+
+    task_folder = task_service.GetFolder(folder_path)
+    task_folder.GetTasks(TASK_ENUM_HIDDEN).each do |task|
+      included = true
```

Comment:
Would be nice to simplify how this reads... how about something like?

```ruby
filter_compatibility = !options[:include_compatibility].empty?
task_folder.GetTasks(TASK_ENUM_HIDDEN).each do |task|
  next if filter_compatibility && !options[:include_compatibility].include?(task.Definition.Settings.Compatibility)
  array << task.Path
end
```

Reply:
Done